bears/general/InvalidLinkBear.py

```diff
@@ -43,8 +43,19 @@ def get_status_code(url, timeout):
     def find_links_in_file(file, timeout, link_ignore_regex):
         link_ignore_regex = re.compile(link_ignore_regex)
         regex = re.compile(
-            r'(https?://[^.:%\s_/?#[\]@\\]+\.(?:[^\s()%\'"`<>|\\]+|'
-            r'\([^\s()%\'"`<>|\\]*\))*)(?<!\.)(?<!,)')
+            r'(https?://'  # http:// or https://
+            r'[^.:%\s_/?#[\]@\\]+'  # Initial Part of Domain
+            r'\.'  # A required dot (.)
+            r'(?:[^\s()%\'\"`<>|\\]+'  # Domain / Subdomain Name
+            r'|'  # OR
+            r'\([^\s()%\'\"`<>|\\]*\))'
+            # Subdomain Name that is contained within ()
+            r'%?[0-9]*'  # A % sign (0 or 1 times) followed by any number
+                        # Thus %20 is allowed and %s isn't allowed
+            r'(?:[^\s()%\'\"`<>|\\]+'  # Same Regex for Domain Name
+            r'|'  # OR
+            r'\([^\s()%\'\"`<>|\\]*\))*)(?<!\.)(?<!,)')  # Same Regex for
```

Comment:
Thanks a lot for the break up.

This now makes it clear that we're using `[^\s()%\'\"`<>|\\]` 4 times !
Can we store it in a variable like `domain_re = r'[^\s()%\'\"`<>|\\]'` and use it to avoid 4 changes if we want to change that regex ?

Note that this is not in the scope of this PR (as you solve the issue). So, preferably, just open a new github issue and mark it as newcomer so that a newcomer can solve it!